arch/RISCV/RISCVInstPrinter.c

```diff
@@ -1,811 +1,416 @@
+		SStream_concat0(O, "-");
+
+		if (ArchRegNames) {
+			SStream_concat0(markup_OS(O, Markup_Register), "x9");
+			SStream_concat0(O, ", ");
+			SStream_concat0(markup_OS(O, Markup_Register), "x18");
+			SStream_concat0(O, "-");
+		}
+		SStream_concat0(
+			markup_OS(O, Markup_Register),
+			doGetRegisterName(
+				RISCV_X19 +
+				(Imm == RISCVZC_RLISTENCODE_RA_S0_S11 ?
+					 8 :
+					 Imm - RISCVZC_RLISTENCODE_RA_S0_S3)));
+		break;
+	default:
+		CS_ASSERT(0 && "invalid register list");
+	}
+	SStream_concat0(O, "}");
+}
+
+void printRegReg(MCInst *MI, unsigned OpNo, SStream *O)
+{
+	RISCV_add_cs_detail_0(MI, RISCV_OP_GROUP_RegReg, OpNo);
+	MCOperand *MO = MCInst_getOperand(MI, (OpNo));
+
+	CS_ASSERT(MCOperand_isReg(MO) &&
+		  "printRegReg can only print register operands");
+	if (MCOperand_getReg(MO) == RISCV_NoRegister)
+		return;
+	printRegName(O, MCOperand_getReg(MO));
+
+	SStream_concat0(O, "(");
+	MCOperand *MO1 = MCInst_getOperand(MI, (OpNo + 1));
+	CS_ASSERT(MCOperand_isReg(MO1) &&
+		  "printRegReg can only print register operands");
+	printRegName(O, MCOperand_getReg(MO1));
+	SStream_concat0(O, ")");
+}
+
+void printSpimm(MCInst *MI, unsigned OpNo, SStream *O)
+{
+	RISCV_add_cs_detail_0(MI, RISCV_OP_GROUP_Spimm, OpNo);
+	int64_t Imm = MCOperand_getImm(MCInst_getOperand(MI, (OpNo)));
+	unsigned Opcode = MCInst_getOpcode(MI);
+	bool IsRV64 = RISCV_getFeatureBits(MI->csh->mode, RISCV_Feature64Bit);
+	bool IsEABI = RISCV_getFeatureBits(MI->csh->mode, RISCV_FeatureRVE);
+	int64_t Spimm = 0;
+	int64_t RlistVal = MCOperand_getImm(MCInst_getOperand(MI, (0)));
+	CS_ASSERT(RlistVal != 16 && "Incorrect rlist.");
+	unsigned Base = RISCVZC_getStackAdjBase(RlistVal, IsRV64, IsEABI);
+	Spimm = Imm + Base;
+	CS_ASSERT((Spimm >= Base && Spimm <= Base + 48) && "Incorrect spimm");
+	if (Opcode == RISCV_CM_PUSH)
+		Spimm = -Spimm;
+
+	RISCVZC_printSpimm(Spimm, markup_OS(O, Markup_Immediate));
+}
+
+void printVMaskReg(MCInst *MI, unsigned OpNo, SStream *O)
+{
+	RISCV_add_cs_detail_0(MI, RISCV_OP_GROUP_VMaskReg, OpNo);
+	MCOperand *MO = MCInst_getOperand(MI, (OpNo));
+
+	CS_ASSERT(MCOperand_isReg(MO) &&
+		  "printVMaskReg can only print register operands");
+	if (MCOperand_getReg(MO) == RISCV_NoRegister)
+		return;
+	SStream_concat0(O, ", ");
+	printRegName(O, MCOperand_getReg(MO));
+	SStream_concat0(O, ".t");
+}
+
+void RISCV_LLVM_printInstruction(MCInst *MI, SStream *O,
+				 void * /* MCRegisterInfo* */ info)
+{
+	MI->MRI = (MCRegisterInfo *)info;
+
+	if (MI->csh->syntax & CS_OPT_SYNTAX_NO_ALIAS_TEXT) {
+		printInstruction(MI, MI->address, O);
+	} else {
+		MCInst Uncompressed;
+		MCInst_Init(&Uncompressed, MI->csh->arch);
+
+		MCInst *McInstr = MI;
+		if (uncompressInst(&Uncompressed, MI)) {
+			McInstr = &Uncompressed;
+			Uncompressed.address = MI->address;
+			Uncompressed.MRI = MI->MRI;
+			Uncompressed.csh = MI->csh;
+			Uncompressed.flat_insn = MI->flat_insn;
+		}
+
+		if (printAliasInstr(McInstr, MI->address, O))
+			MI->isAliasInstr = true;
```

Comment:
Use `MCInst_setIsAlias(MI, isAlias);` instead. Otherwise the member in `cs_insn` is not updated.

Also check `map_use_alias_details` to decide if you have to add real operand details or the ones from the alias. See https://github.com/capstone-engine/capstone/blob/8872be6087dd17734d38f290b119f13a56f3027c/arch/Sparc/SparcInstPrinter.c#L66-L90 how to do it.